src/encoded/types/access_key.py

```diff
@@ -56,33 +65,29 @@ class AccessKey(Item):
     embedded_list = []
 
     STATUS_ACL = {
-        'current': [(Allow, 'role.owner', ['view', 'edit'])] + ONLY_ADMIN_VIEW,
-        'deleted': DELETED,
+        'current': [(Allow, OWNER_ROLE, ['view', 'edit'])] + ONLY_ADMIN_VIEW_ACL,
+        'deleted': DELETED_ACL,
     }
 
     def __ac_local_roles__(self):
         """grab and return user as owner."""
         owner = 'userid.%s' % self.properties['user']
-        return {owner: 'role.owner'}
-
-    def __json__(self, request):
-        """delete the secret access key has from the object when used."""
-        properties = super(AccessKey, self).__json__(request)
-        del properties['secret_access_key_hash']
-        return properties
-
-    def update(self, properties, sheets=None):
-        """smth."""
-        # make sure PUTs preserve the secret access key hash
-        if 'secret_access_key_hash' not in properties:
-            new_properties = self.properties.copy()
-            new_properties.update(properties)
-            properties = new_properties
-        self._update(properties, sheets)
+        return {owner: OWNER_ROLE}
 
     class Collection(Item.Collection):
         pass
 
+# 2023-06-06
+# If these are left to be only defined in snovault and not here then we get on access-key creation (via UI):
+#  File "/Users/dmichaels/.pyenv/versions/3.9.16/envs/snovault-new-3.9.16/lib/python3.9/site-packages/pyramid/viewderivers.py", line 320, in permitted
+#    return authz_policy.permits(context, principals, permission)
+#  File "/Users/dmichaels/.pyenv/versions/3.9.16/envs/snovault-new-3.9.16/lib/python3.9/site-packages/pyramid_localroles/__init__.py", line 86, in permits
+#    principals = local_principals(context, principals)
+#  File "/Users/dmichaels/.pyenv/versions/3.9.16/envs/snovault-new-3.9.16/lib/python3.9/site-packages/pyramid_localroles/__init__.py", line 20, in local_principals
+#    local_roles = local_roles()
+#  File "/Users/dmichaels/repos/cgap/fourfront/src/encoded/types/access_key.py", line 71, in __ac_local_roles__
+#    owner = 'userid.%s' % self.properties['user']
+#KeyError: 'user'
```

Comment:
FYI This is because the view configs are defined inside the type definition file, to do this correctly you would want to have a separate file (so you can run config.include(access_key_views.py)) - see encoded-core for an example of this. I think this is fine though for now.